lib/livebook.ex

```diff
@@ -101,8 +101,28 @@ defmodule Livebook do
       config :logger, level: level
     end
 
-    if metadata = Livebook.Config.log_metadata!("LIVEBOOK_LOG_METADATA") do
-      config :logger, :console, metadata: metadata
+    log_metadata = Livebook.Config.log_metadata!("LIVEBOOK_LOG_METADATA")
+    log_format = Livebook.Config.log_format!("LIVEBOOK_LOG_FORMAT")
+
+    case {log_format, log_metadata} do
+      {:json, nil} ->
+        default_metadata = Application.get_env(:logger, :console, [])[:metadata] || []
+
+        config :logger, :default_handler,
+          formatter: {LoggerJSON.Formatters.Basic, %{metadata: default_metadata}}
+
+      {:json, log_metadata} ->
+        config :logger, :default_handler,
+          formatter: {LoggerJSON.Formatters.Basic, %{metadata: log_metadata}}
+
+      {:text, log_metadata} when not is_nil(log_metadata) ->
+        config :logger, :console, metadata: log_metadata
+
+      {nil, log_metadata} when not is_nil(log_metadata) ->
+        config :logger, :console, metadata: log_metadata
+
+      _ ->
+        :ok
     end
```

Comment:
I think we should be able to write both configs separately like this:

```elixir
if metadata = Livebook.Config.log_metadata!("LIVEBOOK_LOG_METADATA") do
  config :logger, :default_formatter, metadata: metadata
end

case Livebook.Config.log_format!("LIVEBOOK_LOG_FORMAT") do
  :text ->
    :ok
  
  :json ->
    metadata = Application.get_env(:logger, :default_formatter)[:metadata] || []
    config :logger, :default_handler, formatter: {LoggerJSON.Formatters.Basic, metadata: metadata}
end
```

We should also replace current occurrences of `config :logger, :console, ...` with `config :logger, :default_formatter, ...`.

Reply:
I don't think that will work because we have not written to the default formatter yet. This probably will:

```elixir
metadata = Livebook.Config.log_metadata!("LIVEBOOK_LOG_METADATA") || [:request_id]

case Livebook.Config.log_format!("LIVEBOOK_LOG_FORMAT") do
  :text ->
    config :logger, :default_formatter, metadata: metadata

  :json ->
    metadata = Application.get_env(:logger, :default_formatter)[:metadata] || []
    config :logger, :default_handler, formatter: {LoggerJSON.Formatters.Basic, metadata: metadata || []}
end
```

Reply:
I dropped a suggestion above on the version that will definitely work.